extra-dev/packages/coq-rewriter/coq-rewriter.dev/opam

```diff
@@ -13,6 +13,7 @@ build: [
 install: [make "install"]
 depends: [
   "ocaml" {build}
+  "ocaml" {build & >= "4.14.0" & arch != "arm64"}
```

Comment:
I'm a bit confused by the interpretation of `&` here.  It seems that version constraints mean "fail if not satisfied" and non-version constraints mean "ignore if not satisfied", is this right?  Is there a better way to interpret this field?

Reply:
I don't understand why the `arch` part is here at all, I don't think it makes any sense. What people have in the general opam archive is stuff like this:
```
available: [
  (arch != "arm64") &
  (os = "linux" & os-family = "debian")
]
```
Don't you want something similar to this?

Reply:
> Is there a better way to interpret this field?

The official semantics is perform do a two-stage evaluation of the formula. First, you perform a boolean simplification, keeping the version constraints as uninterpreted predicates. Second, you add the remaining version constraints to the solver (or remove the dependency altogether if the formula reduced to `false`). I guess your interpretation is a suitable simplification as long as the formula contains only conjunctions.

If you want OCaml >= 4.14.0 on arm64, then the formula should be
```
"ocaml" {build & (arch != "arm64" | >= "4.14.0")}
```

Reply:
Thanks @silene , I'll change it to that.

@palmskog according to https://github.com/mit-plv/rewriter/issues/97#issuecomment-1453659514, https://github.com/ocaml/ocaml/pull/10595 extended some tail-call optimizations from i386 to all platforms, which landed in 4.14.0.  Without these optimizations, `Search` exhausts the stack when using coq-rewriter files.  So on non-i386 platforms, coq-rewriter requires OCaml >= 4.14.0 to not exhaust the stack.

Reply:
I've swapped to
```coq
"ocaml" {build & (arch = "x86_32" | arch = "x86_64" | >= "4.14.0")}
```
Does this seem right?

Reply:
@JasonGross this looks right if we only consider x86 and arm. But it also means the `>= "4.14.0"` will apply on architectures like `riscv64`. If that's OK, I think it can be merged.

Reply:
Yes, according to https://github.com/ocaml/ocaml/pull/10595, we should expect the issue to occur on all non-i386 platforms, so I want to require>= 4.14.0 on all such platforms